### PR TITLE
hw-24.1

### PR DESCRIPTION
**Pull request by julia-shvabovich**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,9 +4,64 @@
...
+        </maven.checkstyle.plugin.configLocation>
+    </properties>
 
+    <dependencies>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (andriylagun) [inline pom.xml]: Delete this

File: src/main/java/application/model/User.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
+package application.model;
+
+public class User {
+    private Long id;
+    private String name;
+    private String login;
+    private String password;
+
+    public User(Long id, String name, String login, String password) {
+        this.id = id;
```
Comment (andriylagun) [inline src/main/java/application/model/User.java]: Don't need to pass id in constructor. Storage will generate ids

File: src/main/java/application/service/ProductServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    private ProductDao productDao;
+
+    @Override
+    public Product create(Product product) {
+        return productDao.create(product);
+    }
+
+    @Override
+    public Product get(Long id) {
+        return productDao.get(id).orElseThrow();
```
Comment (andriylagun) [inline src/main/java/application/service/ProductServiceImpl.java]: Use get, without orElseThrow

File: src/main/java/application/dao/ProductDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
+package application.dao;
```
Comment (dmytro-hryhoruk) [inline src/main/java/application/dao/ProductDaoImpl.java]:
```suggestion
package application.dao.impl;
```
let's store implementations separate package. Same with with services

File: src/main/java/application/dao/ProductDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+    @Override
+    public Product update(Product product) {
+        List<Product> products = Storage.getProductList();
+        for (int i = 0; i < products.size(); i++) {
+            if (products.get(i).getId().equals(product.getId())) {
+                products.set(i, product);
+            }
+        }
+        return product;
```
Comment (dmytro-hryhoruk) [inline src/main/java/application/dao/ProductDaoImpl.java]: let's do the same with stream

File: src/main/java/application/service/ProductServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
+package application.service;
```
Comment (dmytro-hryhoruk) [inline src/main/java/application/service/ProductServiceImpl.java]:
```suggestion
package application.service.impl;
```

File: src/main/java/application/dao/ProductDaoImpl.java (hunk 4 above)
Comment (julia-shvabovich) [reply src/main/java/application/dao/ProductDaoImpl.java]: done

File: src/main/java/application/service/ProductServiceImpl.java (hunk 6 above)
Comment (julia-shvabovich) [reply src/main/java/application/service/ProductServiceImpl.java]: done

File: src/main/java/application/dao/ProductDaoImpl.java (hunk 5 above)
Comment (julia-shvabovich) [reply src/main/java/application/dao/ProductDaoImpl.java]: done

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**